### DISCUSSION TITLE
Numerous ggml_metal_init messages on init of class

### DISCUSSION THREAD

**Discussion by astrowonk** [abetlen/llama-cpp-python#1885] [Q&A]
I get this when starting up a model class on my M1:

```
gml_metal_init: skipping kernel_get_rows_bf16                     (not supported)
ggml_metal_init: skipping kernel_mul_mv_bf16_f32                   (not supported)
ggml_metal_init: skipping kernel_mul_mv_bf16_f32_1row              (not supported)
ggml_metal_init: skipping kernel_mul_mv_bf16_f32_l4                (not supported)
ggml_metal_init: skipping kernel_mul_mv_bf16_bf16                  (not supported)
ggml_metal_init: skipping kernel_mul_mv_id_bf16_f32                (not supported)
ggml_metal_init: skipping kernel_mul_mm_bf16_f32                   (not supported)
ggml_metal_init: skipping kernel_mul_mm_id_bf16_f32                (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_bf16_h64           (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_bf16_h80           (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_bf16_h96           (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_bf16_h112          (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_bf16_h128          (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_bf16_h256          (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_vec_bf16_h128      (not supported)
ggml_metal_init: skipping kernel_flash_attn_ext_vec_bf16_h256      (not supported)
ggml_metal_init: skipping kernel_cpy_f32_bf16                      (not supported)
ggml_metal_init: skipping kernel_cpy_bf16_f32                      (not supported)
ggml_metal_init: skipping kernel_cpy_bf16_bf16                     (not supported)
```

that was after building with ` CMAKE_ARGS="-DLLAMA_METAL=on" FORCE_CMAKE=1 pip install --upgrade --force-reinstall llama-cpp-python --no-cache-dir `

I'd be content to just run on CPU mode but I get similar warnings if I just do:

`pip install llama-cpp-python --extra-index-url https://abetlen.github.io/llama-cpp-python/whl/cpu`

trying to use the CPU wheel. 

Perhaps these are just normal and I shouldn't worry about it?

**Comment by gyanu2507** [comment]
Those lines are Metal saying "this kernel was compiled out / this dtype is not available on this Apple GPU," not a failed load. `skipping kernel_*_bf16` on an M1 is normal. M1 does not run the bf16 Metal kernels llama.cpp registers for later chips. Your model is almost certainly f16/Q4, so those kernels would never be launched.

`verbose=False` (or `LLAMA_LOG_LEVEL`) quiets the wall of text. It does not change which kernels run.

If the model actually loads and tokens come out, ignore the skip list. Worry only if you then see `failed to load Metal library` or a fallback to CPU with a 10x speed drop.
